ps5Controller/ps5Controller/HomeModule/HomePresenter.swift

```diff
@@ -0,0 +1,102 @@
+import Foundation
+
+protocol HomePresenterInterface {
+    func notifyViewLoaded()
+    func notifyViewDidAppear()
+    func changeTab(type: HomeTabType)
+    func productListFetched(productList: [Product])
+    func productListFetchFailed(with errorMessage: String)
+}
+
+typealias HomeViewModel = (title: String, subtitle: String, image: String)
+typealias Rect = (origin: Point, size: Size)
+typealias Point = (x:Double, y:Double)
+typealias Size = (width: Double, height: Double)
+
+enum HomeTabType: Int {
+    case controller = 0
+    case switcher
+    case mouse
+}
+
+final class HomePresenter: HomePresenterInterface {
+    var view: HomeViewController?
+    var interactor: HomeInteractor?
+    var router: HomeRouter?
```

Comment:
burada şunları şu şekilde düşün, sen presenter'ın init methodunda view'ı, interactor'ı ve router'ı verebilirsin. Bu sayede hem let olurlar hem de nil olmazlar. o yüzden bi tane init methodu eklemen iyi olacak. Sadece view'ın weak olması lazım. Retain cycle oluşmaması için. Altta örnek attım ama advanced bi konu çok da takılma bence. 

kısaca presenter strong olarak view'da tutuluyor, presenter'da da view strong olarak tutulduğunda retain cycle oluşabilir. burada view'ı weak tutarak bunu çözebiliyoruz. PresenterInterface'i yaratmışsın ama view'a o type ile geçmemişsin :( 

Ki burda da interactor ve router'ı da interface ile geçmen lazım. Yani protocollerle erişebileceği methodları kısıtlıyoruz. Her methoda herkes erişemesin diye. Buralara protocollerden erişmen lazım burası çokomelli

https://medium.com/@aks3800/advance-viper-architecture-for-ios-without-retain-cycles-b1995a56903f